src/domain/task/builder/task.builder.ts

```diff
@@ -78,7 +87,17 @@ export class TaskBuilder {
       this.task.lastUpdateDate instanceof Date &&
       this.task.type.length > 0 &&
       this.task.status.length > 0 &&
-      this.task.initiator.length > 0
+      this.task.initiator.length > 0 &&
+      (this.task.progress ? this.progressInvariant(this.task.progress) : true)
+    )
+  }
+
+  private progressInvariant(progress: DeepReadonly<TaskProgress>): boolean {
+    return (
+      progress.current <= progress.max &&
+      progress.current >= progress.min &&
+      progress.max > progress.min &&
+      progress.min < progress.max
```

Comment:
```suggestion
      progress.max > progress.min
```
It is the same condition. Just one is enough

Reply:
of course 🤪